sklearn/cross_decomposition/_pls.py

```diff
@@ -531,8 +531,7 @@ class PLSRegression(_PLS):
     Parameters
     ----------
     n_components : int, default=2
-        Number of components to keep. Should be in `[1, min(n_samples,
-        n_features, n_targets)]`.
+        Number of components to keep. Should be in `[1, n_features]`.
```

Comment:
Going through the math, I think the code is incorrect. The rank of `X.T @ X` is bounded above by `min(n_samples, n_features)` and not just `n_features`. In other words:

```suggestion
        Number of components to keep. Should be in `[1, min(n_samples, n_features)]`.
```

Reply:
Good point, agreed - the rank of a matrix product should be the minimum rank of the original matrices. Should I open an issue for the bug and create a new pull request?

Reply:
> Should I open an issue for the bug and create a new pull request?

Yea, open a bug report and create a new pull request. In the bug report, state that you plan to work on it. Once the new pull request is open, we can close this one. (The documentation should be updated along with the code update).

BTW, you do not need to alter the commit history and force push to your branch. When a PR is merged, we squash the commit so it ends up being one commit anyways.